pkg/validation/validate.go

```diff
@@ -274,13 +274,18 @@ func ValidateProfile(limits ProfileValidationLimits, tenantID string, prof *goog
 			return NewErrorf(MalformedProfile, "function id is 0")
 		}
 	}
+
+	if err := validateStringTableAccess(prof); err != nil {
+		return err
+	}
 	for _, valueType := range prof.SampleType {
 		stt := prof.StringTable[valueType.Type]
```

Comment:
I think this is the only access we need to fix. We duplicate checks done later in pprof validation – it's fine if we merge this, but let's try to avoid duplication. I'm wondering if the sample type check should be performed here, and not in the pprof validation (along with many other checks)

Reply:
I'd like to remove the dupliacate checks, but I am not sure where they are.  If I remove the new validations, all new tests fail. Either return 200 when they should not or panic (sample type / sample label string index oob)

Reply:
Let's merge it as is and sort it out later